### PR TITLE
Initial code to get Box for phar building, replace DeployCommand

### PR DESCRIPTION
**Pull request by stloyd**
Info:
```shell
stloyd@MacBook-Pro-2 static-php-cli % box info spc.phar 

API Version: 1.1.0

Compression: None

Signature: SHA-1
Signature Hash: 127A780E883099614EA82B7BE52EE1E1A15E4E5B

Metadata: None

Contents: 691 files (2.35MB)

```

```shell
stloyd@MacBook-Pro-2 static-php-cli % ./spc.phar                                                                  
static-php-cli 2.0-rc6

Usage:
  command [options] [arguments]

Options:
  -h, --help            Display help for the given command. When no command is given display help for the list command
  -q, --quiet           Do not output any message
  -V, --version         Display this application version
      --ansi|--no-ansi  Force (or disable --no-ansi) ANSI output
  -n, --no-interaction  Do not ask any interactive question
  -v|vv|vvv, --verbose  Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

Available commands:
  build            build CLI binary
  doctor           Diagnose whether the current environment can compile normally
  download         [fetch] Download required sources
  dump-license     Dump licenses for required libraries
  extract          Extract required sources
  help             Display help for a command
  list             List commands
 build
  build:libs       Build dependencies
 dev
  dev:extensions   [list-ext] Helper command that lists available extension details
  dev:php-version  [dev:php-ver] Returns version of PHP located source directory
 micro
  micro:combine    Combine micro.sfx and php code together
```

### CODE REVIEW

**Comment by crazywhalecc** [comment]
After introducing box, it is a good choice to directly rewrite DeployCommand. Ideally, we can automatically package single executable SPC files for different platforms and upload them when releasing a new version:

1. Use the command `bin/spc download --all` to download dependencies
2. Compile the minimally extended phpmicro SAPI required by static-php-cli itself: `bin/spc build pcntl,posix,mbstring,tokenizer,phar --build-micro`
3. Package project: `vendor/bin/box compile`
4. Generate executable binary: `bin/spc micro:combine spc.phar`
5. Upload to release artifacts: `spc-macos-x86_64.tar.gz`

**Comment by stloyd** [comment]
@crazywhalecc Extracted this as a roadmap for rewrite: https://github.com/crazywhalecc/static-php-cli/issues/179

File: .github/workflows/phar-build.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+            - name: "Build PHAR file"
+              run: "composer build:phar"
+
+            - name: "Test PHAR file"
+              run: "./spc.phar dev:extensions"
+
+            - uses: actions/upload-artifact@v3
+              with:
+                  path: spc.phar
```
Comment (crazywhalecc) [inline .github/workflows/phar-build.yml:64]: If it's just for introducing boxes, we don't seem to need the workflow of automatically packaging phar now.

Comment (stloyd) [reply .github/workflows/phar-build.yml:64]: I would leave it, as it's a good base for https://github.com/crazywhalecc/static-php-cli/issues/179, but just remove for now (comment out) autostart on push, etc.

**Comment by crazywhalecc** [comment]
Are there any positive considerations for using the Version class?

**Comment by stloyd** [comment]
It's used by Box as well to mark phar with a given version, the problem it needs to be callable for Box. Do you see any issues with it?

**Comment by stloyd** [comment]
@crazywhalecc Removed the version class, Box was able to call the public const in the end.

**Comment by stloyd** [comment]
Failure unrelated requires a job re-run (probably an issue with GH actions).

**Review by crazywhalecc — APPROVED** [review]

**Pull request merged**